apps/prairielearn/src/components/PageLayout.html.ts

```diff
@@ -0,0 +1,70 @@
+import { html, type HtmlValue } from '@prairielearn/html';
+
+import { HeadContents } from './HeadContents.html.js';
+import { Navbar } from './Navbar.html.js';
+import type { NavbarType, NavPage, NavSubPage } from './Navbar.types.js';
+
+/**
+ * Provides the standard layout for a page, including the head, navbar, and main content.
+ *
+ * @param resLocals The locals object from the Express response.
+ * @param pageTitle The title of the page in the browser.
+ * @param navPage The main page to highlight in the navbar.
+ * @param options.fullWidth Whether the main container should span the entire width of the page.
+ * @param options.marginBottom Whether the main container should have a bottom margin of mb-4 in Bootstrap.
+ * @param options.pageNote A note to display after the pageTitle, shown in parenthesis.
+ * @param options.navSubPage The current subpage, accounted for in the navbar.
+ * @param options.navbarType The type of navbar to render, based on the type of user.
+ * @param headContent Include scripts and other additional head content here.
+ * @param content The main content of the page within the main container.
+ */
+export function PageLayout({
+  resLocals,
+  pageTitle,
+  navPage,
+  options = {},
+  headContent,
+  content,
+}: {
+  resLocals: Record<string, any>;
+  pageTitle: string;
+  navPage: NavPage;
```

Comment:
For all these, I'd advise moving the JS doc comments to the specific properties in the types here. This ensures that Intellisense will show inline documentation for them.

Here's how hovering on `navPage` at a callsite looks now:

<img width="263" alt="Screenshot 2025-01-17 at 14 22 32" src="https://github.com/user-attachments/assets/03ad0b98-41db-409b-a4f6-a7e3df3c409f" />

And here it is after the following change:

```suggestion
  /** The main page to highlight in the navbar. */
  navPage: NavPage;
```

<img width="281" alt="Screenshot 2025-01-17 at 14 22 59" src="https://github.com/user-attachments/assets/09f4c3af-5e7b-4cc2-b3d0-11b47553ae48" />